### PR TITLE
explorer: gas price stats card

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* current reference price
* 7 day average
* historical data for last 7 and 30 epochs

https://user-images.githubusercontent.com/10210143/234885561-1a73d6c1-3a54-42ee-9016-514c60eede14.mov

https://user-images.githubusercontent.com/10210143/234885595-bdd2ebe3-76c2-4811-8a5c-1c3328422e70.mov

NOTE: currently we have to fetch one epoch extra because each epoch's info contain the gas price of the next one. [This PR](https://github.com/MystenLabs/sui/pull/11388) already added the gas price of the epoch itself as well but need to wait until it's deployed (and probably updating sdk as well).

UPDATED TO THIS (did some quick ui changes requested by @mystie711 - the rest will be done [here](https://mysten.atlassian.net/browse/APPS-874))

https://user-images.githubusercontent.com/10210143/234914265-dd6a8290-f124-4b0f-965c-85a6a3f6ac60.mov

closes APPS-829

### CODE REVIEW

File: apps/explorer/src/pages/home/Home.tsx [hunk 1]
```diff
@@ -31,6 +32,8 @@ function Home() {
                 </Suspense>
             </ErrorBoundary>
 
+            <GasPriceCard />
```
Comment (pchrysochoidis) [inline apps/explorer/src/pages/home/Home.tsx:35]: REMOVE Before merging

**Comment by mystie711** [comment]
Also, please shift the dropdown so it aligns exactly on top right with the dropdown
image

File: apps/explorer/src/components/GasPriceCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    const rpc = useRpcClient();
+    return useQuery<EpochGasInfo[]>(
+        ['get', 'last 30 epochs gas price'],
+        async () => {
+            // TODO: update this to get the gas price from the epoch itself rather than the previous one
+            // once this is deployed https://github.com/MystenLabs/sui/pull/11388
+            // every epoch contains the gas price for the next one
+            const epochs = (
+                await rpc.getEpochs({
+                    descendingOrder: true,
```
Comment (Jibz-Mysten) [inline apps/explorer/src/components/GasPriceCard/index.tsx:49]: why order in `descendingOrder` and reverse?

Comment (pchrysochoidis) [reply apps/explorer/src/components/GasPriceCard/index.tsx:49]: To get the latest epochs and then I need them in ascending order for the graph

File: apps/explorer/src/components/GasPriceCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        if (epochsWithPrices.length) {
+            const sum = epochsWithPrices.reduce(
+                (acc, { referenceGasPrice }) =>
+                    acc + BigInt(referenceGasPrice!),
+                0n
+            );
+            return sum / BigInt(epochsWithPrices.length);
+        }
+        return null;
+    }, [data, totalEpochs]);
```
Comment (Jibz-Mysten) [inline apps/explorer/src/components/GasPriceCard/index.tsx:86]: nit 
const data= useHistoricalGasPrices();
...
`return {
       ...data,
        data: average
     };`

File: apps/explorer/src/components/GasPriceCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    );
+    const formattedHoveredPrice = useGasPriceFormat(
+        hoveredElement?.referenceGasPrice ?? null,
+        selectedUnit
+    );
+    const formattedHoveredDate = hoveredElement?.date
+        ? formatDate(hoveredElement?.date, ['month', 'day', 'year'])
+        : '-';
+    return (
+        <Card bg="default" spacing="lg">
```
Comment (Jibz-Mysten) [inline apps/explorer/src/components/GasPriceCard/index.tsx:147]: nit no need for default

**Review by Jibz-Mysten — APPROVED** [review]
LGTM

File: apps/explorer/src/components/GasPriceCard/index.tsx (hunk 3 above)
Comment (pchrysochoidis) [reply apps/explorer/src/components/GasPriceCard/index.tsx:86]: yeah good point, I just wanted to use only loading but I think you are right this is a better pattern!

**Pull request merged**